src/ops/array.rs

```diff
@@ -92,6 +93,25 @@ pub(crate) fn add_by_internal(value: &mut [u32], by: &[u32]) -> u32 {
     carry as u32
 }
 
+pub(crate) fn add_by_internal_flattened(value: &mut [u32; 3], by: u32) -> u32 {
```

Comment:
I previously had something similar - `add_by_internal3`. At some stage I'd love to move away from these functions however it makes sense to introduce it for this PR.